src/argilla/client/workspaces.py

```diff
@@ -13,68 +12,281 @@
     """
 
-    def __init__(self, name: str, *, id: Optional[str] = None) -> None:
-        """Initializes a new `Workspace` instance using the provided name. If no name
-        is provided, the current workspace will be used.
+    client: "httpx.Client"
+    id: UUID
+    name: str
+    users: Optional[List["WorkspaceUserModel"]] = None
+    inserted_at: datetime
+    updated_at: datetime
+
+    def __init__(
+        self, client: "httpx.Client", *, id: Union[str, UUID], name: str, inserted_at: datetime, updated_at: datetime
+    ) -> None:
+        """Initializes a `Workspace` instance. Ideally this class should not be instantiated
+        directly, instead you should use one of the following classmethods:
+            - `Workspace.create` to create a new workspace in Argilla
+            - `Workspace.from_name` to get an existing workspace from Argilla by its name
+            - `Workspace.from_id` to get an existing workspace from Argilla by its ID
 
         Args:
-            name: the name of the workspace to be managed. If not provided, the current
-                workspace will be used.
+            client: the `httpx.Client` initialized to interact with the Argilla API.
+            name: the name of the workspace to be managed.
             id: the ID of the workspace to be managed.
+            inserted_at: the datetime when the workspace was created.
+            updated_at: the datetime when the workspace was last updated.
+
+        Raises:
+            ValueError: if the workspace does not exist in the current Argilla account.
+            ValidationApiError: if the ID provided is not a valid UUID.
+            RuntimeError: if there was an error while retrieving the workspace/s from Argilla.
+
+        Examples:
+            >>> from argilla import rg
+            >>> workspace = rg.Workspace(
+            ...     client=rg.active_client().http_client.httpx,
+            ...     name="my-workspace",
+            ...     id="...",
+            ...     inserted_at=datetime.datetime(2021, 8, 31, 10, 0, 0),
+            ...     updated_at=datetime.datetime(2021, 8, 31, 10, 0, 0),
+            ... )
         """
+        self.client = client
+        self.id = id if isinstance(id, UUID) else UUID(id)
         self.name = name
-        self.id = id or None
+        self.inserted_at = inserted_at
+        self.updated_at = updated_at
 
-        if self.id is None:
-            warnings.warn(
-                f"The `rg.Workspace` doesn't exist yet, you need to create it in advance. Otherwise, use `rg.Workspace.from_name` to make sure that the workspace exists."
+    @property
+    def users(self) -> List["WorkspaceUserModel"]:
+        """Returns the list of users linked to the workspace.
+
+        Returns:
+            A list of `WorkspaceUserModel` instances.
+        """
+        if not hasattr(self, "__users") or self.__users is None:
+            self.__users = workspaces_api.list_workspace_users(self.client, self.id).parsed
```

Comment:
Should be better to create the __users attribute on init, just to know what attributes (public and privates) contain the instances by reading the init method?

Reply:
Since this is just intended to be used internally, we're just defining `users` as that's the property the users will use to list the linked users to a given workspace, there's no need to create `self.__users`, as it won't be accessed/used until `self.users` is accessed.

Reply:
Since this is just intended to be used internally, we're just defining `users` as that's the property the users will use to list the linked users to a given workspace, there's no need to create `self.__users`, as it won't be accessed/used until `self.users` is accessed.